### PR TITLE
So more patches

### PR DESCRIPTION
**Pull request by Skyrimus**
- added Resolution scale factor change support (Full resolution, half resolution)
- fixed some warnings
- added build instructions without Android Studio
- build stability (case sensitive filenames)
- bugfix

### CODE REVIEW

File: Serious-Engine/Sources/Engine/Base/Memory.h [hunk 1]
```diff
@@ -24,19 +24,19 @@ with this program; if not, write to the Free Software Foundation, Inc.,
...
 
 /* Get amount of free memory in system. */
-ENGINE_API extern SLONG GetFreeMemory( void );
+ENGINE_API extern SLLONG GetFreeMemory( void );
 
 /* Allocate a block of memory - fatal error if not enough memory. */
-ENGINE_API extern void *AllocMemory( SLONG memsize );
-ENGINE_API extern void *_debug_AllocMemory( SLONG memsize, int iType, const char *strFile, int iLine);
-ENGINE_API extern void *AllocMemoryAligned( SLONG memsize, SLONG slAlignPow2);
+ENGINE_API extern void *AllocMemory( SLLONG memsize );
```
Comment (aarcangeli) [inline Serious-Engine/Sources/Engine/Base/Memory.h]: Currently the project can be compiled only for a 32 bit cpu.
Is there a particular reason to use a 64 bit type?

File: local.properties [hunk 2]
```diff
@@ -0,0 +1,9 @@
+## This file must *NOT* be checked into Version Control Systems,
+# as it contains information specific to your local configuration.
+#
+# Location of the SDK. This is only used by Gradle.
+# For customization when using a Version Control System, please read the
+# header note.
+#Sat Aug 10 15:21:16 MSK 2019
+ndk.dir=D\:\\androidsdk\\ndk-bundle
+sdk.dir=D\:\\androidsdk
```
Comment (aarcangeli) [inline local.properties]: I think it should be "C:\androidsdk" (as written in the readme file)

File: Serious-Engine/Sources/EntitiesMP/PlayerWeapons.es [hunk 3]
```diff
@@ -644,8 +567,6 @@ components:
   1 class   CLASS_PROJECTILE        "Classes\\Projectile.ecl",
   2 class   CLASS_BULLET            "Classes\\Bullet.ecl",
   3 class   CLASS_WEAPONEFFECT      "Classes\\PlayerWeaponsEffects.ecl",
-  4 class   CLASS_PIPEBOMB          "Classes\\Pipebomb.ecl",
-  5 class   CLASS_GHOSTBUSTERRAY    "Classes\\GhostBusterRay.ecl",
```
[the remark this answers is not in the document]
Comment (Skyrimus) [reply Serious-Engine/Sources/EntitiesMP/PlayerWeapons.es]: In log sends what CPipeBomb and CGhostBusterRay not found

File: Serious-Engine/Sources/Engine/Base/Memory.h (hunk 1 above)
Comment (Skyrimus) [reply Serious-Engine/Sources/Engine/Base/Memory.h]: Just memory improve

File: local.properties (hunk 2 above)
Comment (Skyrimus) [reply local.properties]: No, it's works already

Comment (Skyrimus) [reply local.properties]: Oh, really, fixed

File: Serious-Engine/Sources/EntitiesMP/PlayerWeapons.es (hunk 3 above)
[the remark this answers is not in the document]
Comment (Skyrimus) [reply Serious-Engine/Sources/EntitiesMP/PlayerWeapons.es]: And it not used in game

File: local.properties (hunk 2 above)
Comment (Skyrimus) [reply local.properties]: I have dualboot 32 bit and 64 bit roms on device and can test it on different OS

Comment (Skyrimus) [reply local.properties]: This a little bit of my changes, more features not commited yet

**Comment by Skyrimus** [comment]
I think will be better, if use Android.mk to compile, because it will fix build on linux

**Comment by Skyrimus** [comment]
Maybe you have some ideas to fix build using linux? I tried to fix it, but have forward declaration error while using classes.

**Comment by aarcangeli** [comment]
I've never tried to build it on Linux.
The problem is that ecc.exe cannot be run  on linux, did you built for linux it in your own?

**Comment by aarcangeli** [comment]
sorry, in the end I decided to remove local.properties in order to simplify the build on android studio

**Comment by Skyrimus** [comment]
Ok, it not important :D

**Comment by Skyrimus** [comment]
I have ecc on linux, but compiler give error forward declarations of classes

**Pull request merged**